### PR TITLE
Add widgets documentation to tardis docs

### PR DESCRIPTION
**Pull request by jaladh-singhal**
It adds documentation of widgets to tardis docs by creating a directory `tardis_widgets` having:
- `using_widgets.rst` - For description about how to use widgets and their features
- `generating_widgets.ipynb` - Demo notebook to generate widgets

> Note: This PR is branched on PR#1264, so should only be merged after #1264 gets merged

### CODE REVIEW

File: docs/running/tardis_widgets/using_widgets.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+over to `Generating TARDIS Widgets <generating_widgets>`_ section and see the
+notebook in action.
+
+Currently, TARDIS supports the following widgets:
+
+Shell Info Widget
+#################
+
+This widget allows you to explore the chemical abundances in different shells
+of a Supernova ejecta.
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:16]: "of the model supernova ejecta"

File: docs/running/tardis_widgets/using_widgets.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+.. image:: images/shell-info-widget-demo.gif
+    :alt: Demo of Shell Info Widget
+
+It consists of four interlinked tables - clicking on any row in a table,
+populates data in the table(s) right to it. All these 4 tables, in order, has
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:22]: "tables to the right of it"

"Each table has the following information"

File: docs/running/tardis_widgets/using_widgets.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+following information:
+
+1. **Shells Data** - Radiative temperature and Dilution Factor (W) of each shell
+(computational partitions) of a Supernova ejecta. Shell numbers in ascending
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:26]: "for each cell...of the model supernova ejecta"

"Shell numbers are given in ascending order, from the innermost shell to the outermost."

File: docs/running/tardis_widgets/using_widgets.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+4. **Level Abundances** - Fractional abundance of each level of the selected
+ion and element in the selected shell.
+
+.. any more info about ion no.s (0 means no charge and hence I) and level no.s
+
+Line Info Widget
+################
+
+This widget lets you explore line interactions present in the spectrum of a
+simulation model.
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:45]: "explore the atomic lines responsible for producing features in the simulated spectrum"

File: docs/running/tardis_widgets/using_widgets.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+It consists of four interlinked tables - clicking on any row in a table,
+populates data in the table(s) right to it. All these 4 tables, in order, has
+following information:
+
+1. **Shells Data** - Radiative temperature and Dilution Factor (W) of each shell
+(computational partitions) of a Supernova ejecta. Shell numbers in ascending 
+order represent inner to outer i.e. innermost shell is numbered as 1 and 
+outermost shell is numbered as the last number (say 20).
+
+2. **Element Abundances** - Fractional abundance of each element present in the
```
Comment (ycamacho) [inline docs/running/tardis_widgets/using_widgets.rst:30]: Fractional (mass) abundance

File: docs/running/tardis_widgets/using_widgets.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+################
+
+This widget lets you explore line interactions present in the spectrum of a
+simulation model.
+
+.. image:: images/shell-info-widget-demo.gif
+    :alt: Demo of Line Info Widget
+
+You can select any wavelength range in the spectrum interactively to display a 
+table of species abundances (fraction of packets interacting). Using
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:51]: "display a table giving the fraction of packets that experienced their last interaction with each element."

File: docs/running/tardis_widgets/using_widgets.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+This widget lets you explore line interactions present in the spectrum of a
+simulation model.
+
+.. image:: images/shell-info-widget-demo.gif
+    :alt: Demo of Line Info Widget
+
+You can select any wavelength range in the spectrum interactively to display a 
+table of species abundances (fraction of packets interacting). Using 
+toggle buttons, you can specify whether to filter the selected range by
+emitted or absorbed wavelengths of packets. Clicking on a row in species
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:53]: "by the emitted..."

File: docs/running/tardis_widgets/using_widgets.rst [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+1. **Shells Data** - Radiative temperature and Dilution Factor (W) of each shell
+(computational partitions) of a Supernova ejecta. Shell numbers in ascending 
+order represent inner to outer i.e. innermost shell is numbered as 1 and 
+outermost shell is numbered as the last number (say 20).
+
+2. **Element Abundances** - Fractional abundance of each element present in the
+selected shell.
+
+3. **Ion Abundances** - Fractional abundance of each ion (species) of the
```
Comment (ycamacho) [inline docs/running/tardis_widgets/using_widgets.rst:33]: Fractional (mass) abundance

File: docs/running/tardis_widgets/using_widgets.rst [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    :alt: Demo of Line Info Widget
+
+You can select any wavelength range in the spectrum interactively to display a 
+table of species abundances (fraction of packets interacting). Using 
+toggle buttons, you can specify whether to filter the selected range by
+emitted or absorbed wavelengths of packets. Clicking on a row in species
+abundances table further reveals packet counts for each last line 
+interaction which can be grouped by excitation lines, de-excitation lines
+or both, using the dropdown.
+
```
Comment (jordi5) [inline docs/running/tardis_widgets/using_widgets.rst:57]: Clicking on a row in "species abundance" table reveals packet counts for each last line interaction. These packet counts can be grouped by excitation lines, de-excitation lines or both, using the dropdown (arrow?, menu?).

File: docs/running/tardis_widgets/using_widgets.rst [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+order represent inner to outer i.e. innermost shell is numbered as 1 and 
+outermost shell is numbered as the last number (say 20).
+
+2. **Element Abundances** - Fractional abundance of each element present in the
+selected shell.
+
+3. **Ion Abundances** - Fractional abundance of each ion (species) of the
+selected element present in the selected shell. 
+
+4. **Level Abundances** - Fractional abundance of each level of the selected
```
Comment (ycamacho) [inline docs/running/tardis_widgets/using_widgets.rst:36]: Fractional (mass) abundance

File: docs/running/tardis_widgets/using_widgets.rst [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+=========================
+The spectrum in line info widget is an interactive figure made using plotly,
+there are several things you can do with it:
+
+Making Selection
+----------------
+By default, the box selection is enabled so you just need to click and drag on
+figure and the pink colored selection box will appear. If you're wondering
+why it has full height, it's because we're only interested in controlling
+widths to specify range on X-axis i.e. wavelength range.
```
Comment (jordi5) [inline docs/running/tardis_widgets/using_widgets.rst:71]: ... specify range on the wavelength axis

File: docs/running/tardis_widgets/using_widgets.rst [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+Interacting with Spectrum
+=========================
+The spectrum in line info widget is an interactive figure made using plotly,
+there are several things you can do with it:
+
+Making Selection
+----------------
+By default, the box selection is enabled so you just need to click and drag on
+figure and the pink colored selection box will appear. If you're wondering
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:69]: "drag on the figure and a pink"

I would remove the next sentence and just put: "Make a selection across the wavelength range you're interested in and the line info table will update to show only packets from this region"

File: docs/running/tardis_widgets/using_widgets.rst [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+.. image:: images/spectrum_selection.gif
+
+After making a selection, if you need to resize the selection box (say make it
+narrower), simply redraw a new selection box over the older one.
+
+
+Using Rangesilder
+-----------------
+Since spectra are usually very long, so it makes sense to focus (horizontally
```
Comment (jordi5) [inline docs/running/tardis_widgets/using_widgets.rst:81]: Use either "since" or "so" using both is redundant.

Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:81]: I would change this paragraph to "The rangeslider is a long bar below the figure that allows you to zoom into particular wavelength regions."

File: docs/running/tardis_widgets/using_widgets.rst [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+Either you can **slide** the focused range by clicking and dragging it or you 
+can **resize** it by dragging the handles (vertical bars) at its edges.
+
+Using other options in Modebar
+------------------------------
+If you take your mouse to the top right corner of figure, you will see a Modebar
+with multiple options. The default option when line info widget first displays
+is ***Box Select*** - the dotted square icon. You can click on other options
+like ***Zoom** (magnifying glass icon), to do a rectangle zoom which maybe
+helpful to focus on a feature in spectrum. You can always revert back to
```
Comment (jordi5) [inline docs/running/tardis_widgets/using_widgets.rst:96]: feature in *the* spectrum

File: docs/running/tardis_widgets/generating_widgets.ipynb [hunk 15]
```diff
@@ -4,18 +4,24 @@
    "cell_type": "markdown",
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/generating_widgets.ipynb:4]: <p>"TARDIS widgets that allows..."</p><p> </p><p>"This notebook represents a quick-start tutorial, but more detail on each widget (and its features) is giving in the Using TARDIS Widgets section of the documentation"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1280/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/generating_widgets.ipynb:4]: <p>TARDIS, for consistency</p><p> </p><p>"as an HDF"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1280/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/generating_widgets.ipynb:4]: <p>By selecting a wavelength range on the spectrum plot, you can see the species responsible for producing the features within that range. This is determined by counting the number of escaping packets that experienced their last interaction with each ion. Packets can be filtered by the wavelength at which they were absorbed or emitted. You can then click on each species to see the packets counts for specific lines. These can be grouped in several ways. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1280/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: docs/running/tardis_widgets/using_widgets.rst [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+.. image:: ../images/line_info_spectrum_selection.gif
+    :alt: Demo of making selection
+
+After making a selection, if you need to resize the selection box (say, make it
+narrower), simply redraw a new selection box over the older one.
+
+Using Rangesilder
+-----------------
+The rangeslider is a long bar below the figure that allows you to zoom on a
```
Comment (jamesgillanders) [inline docs/running/tardis_widgets/using_widgets.rst:79]: ...to zoom in on a...

File: docs/running/tardis_widgets/using_widgets.rst [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+Either you can **slide** the zoomed range by clicking and dragging it or you 
+can **resize** it by dragging the handles (vertical bars) at its edges.
+
+Using other options in Modebar
+------------------------------
+If you take your mouse to the top right corner of figure, you will see a Modebar
+with multiple options. The default option when line info widget first displays
+is **Box Select** - the dotted square icon. You can click on other options
+like **Zoom** (magnifying glass icon), to do a rectangle zoom which maybe
+helpful to focus on a feature in the spectrum. You can always revert back to
```
Comment (jamesgillanders) [inline docs/running/tardis_widgets/using_widgets.rst:94]: ... a rectangular zoom...

Comment (jamesgillanders) [reply docs/running/tardis_widgets/using_widgets.rst:94]: which may be helpful

**Review by jamesgillanders — CHANGES_REQUESTED** [review]
great work @jaladh-singhal! Very well documented. Happy for this to be merged, pending review of my comments

**Review by jamesgillanders — APPROVED** [review]

File: docs/running/tardis_widgets/using_widgets.rst [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+#################
+
+This widget allows you to explore the chemical abundances in different shells
+of the model Supernova ejecta.
+
+.. image:: ../images/shell_info_widget_demo.gif
+    :alt: Demo of Shell Info Widget
+
+It consists of four interlinked tables - clicking on any row in a table,
+populates data in the table(s) right of it. Each table has the following
```
Comment (MarkMageeAstro) [inline docs/running/tardis_widgets/using_widgets.rst:22]: ...table(s) to the right of it.

**Review by MarkMageeAstro — CHANGES_REQUESTED** [review]
One minor comment regarding grammar, but otherwise looks good.

**Review by MarkMageeAstro — APPROVED** [review]
Looks all good to me.

**Review by marxwillia — APPROVED** [review]

**Pull request merged**